astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
+        10- 14  E5.1   ---    e       [-3160000.0/0.01] Description of e
+        16- 23  F8.5   ---    d       [22.25/27.25] Description of d
+        25- 31  E7.1   ---    s       [-9e+34/2.0] Description of s
+        33- 35  I3     ---    i       [-30/67] Description of i
+        37- 39  F3.1   ---    sameF   [5.0/5.0] Description of sameF
+        41- 42  I2     ---    sameI   [20] Description of sameI
+        44- 47  F4.1   h      RAh     Right Ascension (hour)
+        49- 51  F3.1   min    RAm     Right Ascension (minute)
+        53- 70  F18.15 s      RAs     Right Ascension (second)
+            72  A1     ---    DE-     Sign of Declination
+        73- 76  F5.1   deg    DEd     Declination (degree)
+        78- 81  F4.1   arcmin DEm     Declination (arcmin)
+        83- 98  F16.13 arcsec DEs     Declination (arcsec)
+
+        --------------------------------------------------------------------------------
+        """
+        # Get column widths
+        vals_list = []
+        col_str_iters = self.data.str_vals()
+        for vals in zip(*col_str_iters):
+            vals_list.append(vals)
+
+        for i, col in enumerate(self.cols):
+            col.width = max([len(vals[i]) for vals in vals_list])
+            if self.start_line is not None:
+                col.width = max(col.width, len(col.info.name))
+        widths = [col.width for col in self.cols]
+
+        startb = 1  # Byte count starts at 1.
+
+        # Set default width of the Bytes count column of the Byte-By-Byte table.
+        # This ``byte_count_width`` value helps align byte counts with respect
+        # to the hyphen using a format string.
+        byte_count_width = len(str(sum(widths) + len(self.cols) - 1))
+
+        # Format string for Start Byte and End Byte
+        singlebfmt = "{:" + str(byte_count_width) + "d}"
+        fmtb = singlebfmt + "-" + singlebfmt
+        # Add trailing single whitespaces to Bytes column for better visibility.
+        singlebfmt += " "
+        fmtb += " "
+
+        # Set default width of Label and Description Byte-By-Byte columns.
+        max_label_width, max_descrip_size = 7, 16
+
+        bbb = Table(names=['Bytes', 'Format', 'Units', 'Label', 'Explanations'],
+                    dtype=[str]*5)
+
+        # Iterate over the columns to write Byte-By-Byte rows.
+        for i, col in enumerate(self.cols):
+            # Check if column is MaskedColumn
+            col.has_null = isinstance(col, MaskedColumn)
+
+            # Check if the column format was set by the passed ``formats`` argument.
+            if col.format is None:
+                if col.info.format is not None:
+                    col.format = col.info.format
+
+            if col.format is not None:
+                col.meta.size = max([len(sval) for sval in col.str_vals])
+
+            # Set CDSColumn type, size and format.
+            if np.issubdtype(col.dtype, np.dtype(int).type) or np.issubdtype(col.dtype, np.integer):
```

Comment:
Was this a result of the ``DeprecationWarning`` issue @Suyog7130?

Reply:
Yeah, it was because of that. However, having both the conditions is not necessary.

Reply:
They don't hurt and can be removed once we drop (I think) numpy 1.18, which will happen in a few months time. However, we don't want to wait this long with merging since we hope to get through all of @Suyog7130 PRs by then! So for, now this (or a similar) crutch needs to stay in.